Foundation/URLCache.swift

```diff
@@ -142,13 +147,37 @@ open class URLCache : NSObject {
     */
     open class var shared: URLCache {
         get {
-            NSUnimplemented()
+            return sharedSyncQ.sync {
+                if let cache = sharedCache {
+                    return cache
+                } else {
+                    let fourMegaByte = 4 * 1024 * 1024
+                    let twentyMegaByte = 20 * 1024 * 1024
+                    let bundleIdentifier = Bundle.main.bundleIdentifier ?? UUID().uuidString
+                    let cacheDirectoryPath = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first?.path ?? "\(NSHomeDirectory())/Library/Caches"
```

Comment:
Same idea here. If we don’t find a good URL for `.cachesDirectory` maybe it is better to not have disk cache.

The fallback has the problem of being very Darwin specific. For example `NSHomeDirectory()` in Android is the root directory, where one cannot write at all. Also, it might want to use `appendingPathComponent(…).appendingPathComponent(…)` to be nicer to Windows.

(There's another case below where the POSIX path separator is used instead of a higher level function).

Reply:
The home directory should be in a persistent writable path everywhere, much as possible. I know that Android is _really_ weird about this, but that looks like incorrect mapping to me.

Reply:
Yeah, please do not fall back like this. If `FileManager` does not return a caches URL, consider not caching at all.

Reply:
Sure, I will remove the fallback if `FileManager` for `cachesDirectory` returned nill.